### ISSUE TITLE
pact-js vs. pact-node

### ISSUE THREAD

**Issue by basickarl** [pact-foundation/pact-js#408]
What is the difference between the pact-js, pact-node and pact-web packages?

We have the following types of code:
- asynchronous
- synchronous
- node.js
- browser
- isomorphic

Which package should one choose if doing one of the following in my list?

**Comment by mefellows** [comment]
TL;DR - you want Pact JS in most instances.

* asynchronous - [Pact JS](https://www.npmjs.com/package/@pact-foundation/pact)
* synchronous - [Pact JS](https://www.npmjs.com/package/@pact-foundation/pact)
* node.js - [Pact JS](https://www.npmjs.com/package/@pact-foundation/pact)
* browser - If you _actually_ need it to run in the browser (almost always debateable), [Pact Web](https://www.npmjs.com/package/@pact-foundation/pact-web) 
* isomorphic - If you _actually_ need it to run in the browser (almost always debateable), [Pact Web](https://www.npmjs.com/package/@pact-foundation/pact-web) 

The "I need to run it in the browser" question comes up occasionally. The question is this - for your JS code to be able to make a call to another API, is this dependent on browser-specific code? In most cases, people use tools like React/Angular which have libraries that work on the server and client side, in which case, these tests don't need to run in a browser. 

I think this table is a good one, and I'm going to use it as a starting point to publish onto the various repositories to make things clearer.

See also https://github.com/pact-foundation/pact-node/issues/145.

This from the Pact Node github home page:

<img width="979" alt="Screen Shot 2020-02-07 at 8 13 13 am" src="https://user-images.githubusercontent.com/53900/73979170-ca306700-4981-11ea-9c87-915ef72e3b0b.png">

Does this answer your question?

**Comment by basickarl** [comment]
@mefellows Perfect! Thank you very muchos!

**Comment by mefellows** [comment]
You're welcome. I've just added a first cut at a table. Not sure it's in the right place but it's a start.
